webapp/src/components/common/EditableMatrix/index.tsx

```diff
@@ -0,0 +1,116 @@
+import { useEffect, useState, useRef } from "react";
+import _ from "lodash";
+import HotTable, { HotColumn } from "@handsontable/react";
+import { registerAllModules } from "handsontable/registry";
+// eslint-disable-next-line import/no-unresolved
+import { CellChange } from "handsontable/common";
+import { MatrixIndex, MatrixEditDTO, MatrixType } from "../../../common/types";
+import "handsontable/dist/handsontable.min.css";
+import MatrixGraphView from "./MatrixGraphView";
+import { Root, StyledHotTable } from "./style";
+import "./style.css";
+import { createDateFromIndex, slice } from "./utils";
+
+interface PropTypes {
+  matrix: MatrixType;
+  matrixIndex?: MatrixIndex;
+  matrixTime: boolean;
+  readOnly: boolean;
+  toggleView?: boolean;
+  onUpdate?: (change: MatrixEditDTO[], source: string) => void;
+}
+
+type CellType = Array<number | string | boolean>;
+type ColumnsType = { title: string; readOnly: boolean };
+
+function EditableMatrix(props: PropTypes) {
+  const { readOnly, matrix, matrixIndex, matrixTime, toggleView, onUpdate } =
+    props;
+  const { data = [], columns = [], index = [] } = matrix;
+  const prependIndex = index.length > 0 && matrixTime;
+  const [grid, setGrid] = useState<Array<CellType>>([]);
+  const [formatedColumns, setColumns] = useState<Array<ColumnsType>>([]);
+  const hotTableComponent = useRef<HotTable>(null);
+
+  registerAllModules();
```

Comment:
Move it in the entry point file of the application, outside a function.
It's better to register only the modules used https://handsontable.com/docs/react-modules/#step-1-import-the-handsontable-base-module

Reply:
By entry point file I mean _**src/index.tsx**_